Add support for downloading remote images

designbymind opened the pull request:
Additional expectingImage param for extraParams object.

designbymind committed:
Add support for downloading remote images

Adding additional expectingImage property to extraParams.

designbymind committed:
Merge pull request #1 from designbymind/designbymind-patch-1

Add support for downloading remote images

Topener commented:
Hi! Thanks for your contribution, it looks like this can be used for any filetype, not just images. Perhaps change it to a more generic implementation? `expectRaw` for example?

designbymind committed:
Rename expectingImage to expectingFile

Since responseData could be filetype other than image, let's use expectingFile instead (of expectingImage).

designbymind committed:
Add extraParams for error handling

Add extraParams param to each method for error handling

designbymind committed:
Update ti.xhr.js

designbymind committed:
Update ti.xhr.js

Topener commented:
Ok looking good! Could you bring in the changes in the recent update? Then I can go ahead and merge it!
